components/governance/org.wso2.carbon.governance.registry.extensions/src/org/wso2/carbon/governance/registry/extensions/executors/utils/Utils.java

```diff
@@ -190,4 +199,40 @@ public static List evaluateXpath(OMElement contentElement, String xpathString) {
         return null;
     }
 
+	/**
+	 * Authenticate to API Manager
+	 *
+	 * @param httpContext HTTP context.
+	 */
+	public static void authenticateAPIM(HttpContext httpContext, String apimEndpoint, String apimUsername,
+	                                    String apimPassword) throws RegistryException {
+		String loginEP = apimEndpoint + ExecutorConstants.APIM_LOGIN_URL;
+		try {
+			// create a post request to addAPI.
+			HttpClient httpclient = new DefaultHttpClient();
+			HttpPost httppost = new HttpPost(loginEP);
+			// Request parameters and other properties.
+			List<NameValuePair> params = new ArrayList<NameValuePair>(3);
+
+			params.add(new BasicNameValuePair(ExecutorConstants.API_ACTION, ExecutorConstants.API_LOGIN_ACTION));
+			params.add(new BasicNameValuePair(ExecutorConstants.API_USERNAME, apimUsername));
+			params.add(new BasicNameValuePair(ExecutorConstants.API_PASSWORD, apimPassword));
+			httppost.setEntity(new UrlEncodedFormEntity(params, "UTF-8"));
+
+			HttpResponse response = httpclient.execute(httppost, httpContext);
+			if (response.getStatusLine().getStatusCode() != 200) {
```

Comment:
Its better to use response codes as in the doc [1] rather than just put 200

[1] - https://wink.apache.org/documentation/1.0/api/org/apache/wink/common/http/HttpStatus.html